### ISSUE TITLE
Add support to compile task on Gradle with Java 10

### ISSUE THREAD

**Issue by jeandersonbc** [javapathfinder/jpf-core#139]
Related to #138 

Currently unable to compile classes but I haven't added compile args at the moment.

**Comment by jeandersonbc** [comment]
@gayanW I was checking these changes for reference: https://github.com/javapathfinder/jpf-core/pull/28

I made some progress regarding those "package ... is declared in module ... which does not export it to unanmed module)"-like errors but I hitting another compilation error:

```
/tmp/src/classes/gov/nasa/jpf/SerializationConstructor.java:23: error: cannot inherit from final Constructor
public class SerializationConstructor<T> extends Constructor<T> {
                                                 ^
/tmp/src/classes/gov/nasa/jpf/SerializationConstructor.java:23: error: constructor Constructor in class Constructor<T#2> cannot be applied to given types;
public class SerializationConstructor<T> extends Constructor<T> {
       ^
  required: Class<T#1>,Class<?>[],Class<?>[],int,int,String,byte[],byte[]
  found: no arguments
  reason: actual and formal argument lists differ in length
  where T#1,T#2 are type-variables:
    T#1 extends Object declared in class SerializationConstructor
    T#2 extends Object declared in class Constructor
2 errors
```

The class `Constructor` from `src/classes/modules/java.base/java/lang/reflect/Constructor.java` is right. Therefore, I'm guessing this error is happening because the module `java.base` has to be compiled first.

Did you have this error when you were doing this change?

**Comment by gayanW** [comment]
It is needed that we patch java.lang.reflect.Constructor with our MJI model class for Constructor, which is not `final`.

```
<compilerarg value="--patch-module"/>
<compilerarg value="java.base=src/classes/modules/java.base"/>
```

**Comment by jeandersonbc** [comment]
Great. I'm still getting how Jigsaw works but feels like we are in the right direction...

**Comment by jeandersonbc** [comment]
Still struggling to compile the modules on Gradle. I tried some alternatives from https://guides.gradle.org/building-java-9-modules but didn't work as expected.

It is like adding `--patch-module` doesn't have any effect.
I tried to compile the modules first but, again, it didn't work.
I need to debug the Gradle execution to understand what's happening because I'm able to compile the modules via command-line with the JDK only.

``
